### PR TITLE
Adding Delete Account Functionality to the Settings Page

### PR DESCRIPTION
**Pull request by gkatrakazas**
This pull request adds a "Delete Account" functionality to the Settings Page, providing users with greater control over their accounts.

Connected with: https://github.com/wwWallet/wallet-backend-server/pull/46

### CODE REVIEW

**Review by kkmanos — APPROVED** [review]

File: src/locales/en.json [hunk 1]
```diff
@@ -118,21 +118,34 @@
...
+			"description": "Once you delete your account, there is no going back. Please be certain.",
+			"messageDeletePasskeyPart1": "Are you sure you want to delete",
+			"messageDeletePasskeyPart2": "passkey?",
+			"messageDeleteAccount": "Are you sure you want to delete <strong> your Account </strong>?",
+			"title": "Delete account"
+		},
+		"deletePasskeyButtonTitleLocked":"You cannot add passkey while locked",
 		"description": "View account information and manage passkeys",
 		"lockPasskeyManagement": "Lock passkey management",
+		"noOtherPasskeys": "There is no other passkeys in this account.",
```
Comment (emlun) [inline src/locales/en.json:133]:
```suggestion
		"noOtherPasskeys": "There are no other passkeys in this account.",
```

File: src/locales/en.json [hunk 2]
```diff
@@ -146,10 +159,12 @@
 			"passkeyCreated": "Your passkey has been created."
 		},
 		"title": {
+			"confirmDeletePopup": "Are You Sure?",
 			"loggedInPasskey": "Logged in Passkey",
+			"manageAcount": "Manage Acount",
```
Comment (emlun) [inline src/locales/en.json:164]:
```suggestion
			"manageAcount": "Manage Account",
```

File: src/pages/Settings/Settings.tsx [hunk 3]
```diff
@@ -363,16 +365,35 @@ const WebauthnCredentialItem = ({
 	credential,
 	onDelete,
 	onRename,
+	unlocked
+
 }: {
 	credential: WebauthnCredential,
 	onDelete?: false | (() => void),
```
Comment (emlun) [inline src/pages/Settings/Settings.tsx:372]: This type annotation should be updated now that we've added an `await` on `onDelete()` below.
```suggestion
	onDelete?: false | (() => Promise<void>),
```

Comment (emlun) [reply src/pages/Settings/Settings.tsx:372]: Note also that `deleteWebauthnCredential` also calls `refreshData()` at the end, but without `await` - meaning that `await onDelete()` will _not_ wait for `refreshData()` to complete. If we want `onDelete()` to also wait for `refreshData()` to complete, we need to add an `await` to `refreshData()` as well (there are a few more occurrences like this that could use review too).

File: src/locales/en.json [hunk 4]
```diff
@@ -118,21 +118,34 @@
 	},
 	"pageSettings": {
 		"addPasskey": "Add passkey",
+		"deleteAccount": {
+			"buttonText": "Delete your account",
+			"deleteButtonTitleLocked":"You cannot delete your account while locked",
+			"description": "Once you delete your account, there is no going back. Please be certain.",
+			"messageDeletePasskeyPart1": "Are you sure you want to delete",
+			"messageDeletePasskeyPart2": "passkey?",
+			"messageDeleteAccount": "Are you sure you want to delete <strong> your Account </strong>?",
```
Comment (emlun) [inline src/locales/en.json:127]: These are all unused, instead the messages are hard-coded:

- [`<ConfirmDeletePopup>` to delete passkey](https://github.com/wwWallet/wallet-frontend/pull/167/files#diff-9760b7babfb1eae127ff9f01f25c6494cd24d2e638591ebfb83db7e8885fadb0R534)
- [`<ConfirmDeletePopup>` to delete account](https://github.com/wwWallet/wallet-frontend/pull/167/files#diff-9760b7babfb1eae127ff9f01f25c6494cd24d2e638591ebfb83db7e8885fadb0R742)

**Pull request merged**